### PR TITLE
Table view dynamic cells

### PR DESCRIPTION
**Pull request by Cona7**

### CODE REVIEW

File: ZeplinTask/EditViewController.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import UIKit
+
+class EditViewController: UITableViewController {
+
+    struct CellModel {
+
+        let name: String
+        let number: String = "8545"
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: `: String` nije potreban

File: ZeplinTask/EditViewController.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    let pictureString = "Image-1"
+
+    let arrayTotal: [CellModel] = [CellModel(name: "Total workouts"),
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: `: [CellModel]` nije potreban.

Za array je bolje staviti element u novi red ako ima vise od 1 element

```
[
    CellModel(name: "Total workouts"),
    CellModel(name: "Total reps")
]
```

File: ZeplinTask/EditViewController.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    let arrayMuscles: [CellModel] = [CellModel(name: "Triceps"),
+                                     CellModel(name: "Biceps"),
+                                     CellModel(name: "Bench")]
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let menuBtn = UIButton(type: .custom)
+        menuBtn.frame = CGRect(x: 0.0, y: 0.0, width: 20, height: 20)
+        menuBtn.setImage(UIImage(named: pictureString), for: .normal)
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Zamjeni `UIImage(named: )` s `Image Literal`. To je noviji i "compile safe" nacin dodavanja slika.

File: ZeplinTask/EditViewController.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        menuBtn.addTarget(self, action: #selector(editButtonTapped), for: UIControl.Event.touchUpInside)
+
+        let settingsBarItem = UIBarButtonItem(customView: menuBtn)
+        let currWidth = settingsBarItem.customView?.widthAnchor.constraint(equalToConstant: 22.3)
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Fix image (1, 2, 3x).

Remove constraints and frame from BarButton.

File: ZeplinTask/EditViewController.swift [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        currWidth?.isActive = true
+        let currHeight = settingsBarItem.customView?.heightAnchor.constraint(equalToConstant: 22.3)
+        currHeight?.isActive = true
+        self.navigationItem.rightBarButtonItem = settingsBarItem
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Remove `self.`

File: ZeplinTask/EditViewController.swift [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            if let navigator = navigationController {
+                navigator.pushViewController(editProfileViewController, animated: true)
+            }
+        }
+    }
+}
+
+extension EditViewController {
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Func not needed.

File: ZeplinTask/EditViewController.swift [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return true
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 0 {
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: `return section == 0 ? arrayTotal.count  : arrayMuscles.count` is better for this use case IMO

File: ZeplinTask/EditViewController.swift [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath) as? TypeViewCell else {
+            fatalError("The dequeued cell is not an instance of TypeViewCell.")
+        }
+
+        let workoutCell = indexPath.section == 0 ? arrayTotal[indexPath.row] : arrayMuscles[indexPath.row]
+
+        cell.stringLabel.text = workoutCell.name
+        cell.numberLabel.text = workoutCell.number
+        cell.numberLabel.textColor = .customBlue
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Can the text font and color be set in IB?

File: ZeplinTask/EditProfileViewController.swift [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    func initUISegmentedControl(_ segmentedControl: UISegmentedControl) {
+        segmentedControl.tintColor = .customBlue
+        segmentedControl.layer.cornerRadius = 4
+    }
+
+    @objc
+    func keyboardWillShow(notification: Notification) {
+        if let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
```
Comment (Rep2) [inline ZeplinTask/EditProfileViewController.swift]: Can the 2 ifs be combined?

File: ZeplinTask/EditProfileViewController.swift [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+            if keyboardSize.origin.y < (weightTextField.frame.origin.y + weightTextField.frame.height) {
+                let constantToRaiseConstraint = keyboardSize.origin.y - weightTextField.frame.origin.y - weightTextField.frame.height - 20
+                imageViewTopConstraint.constant += constantToRaiseConstraint
+                buttonBottomConstraint.constant -= constantToRaiseConstraint
+                self.view.layoutIfNeeded()
```
Comment (Rep2) [inline ZeplinTask/EditProfileViewController.swift]: Unneeded `self.`

**Review by Rep2 — CHANGES_REQUESTED** [review]

File: ZeplinTask/EditViewController.swift (hunk 8 above)
Comment (Cona7) [reply ZeplinTask/EditViewController.swift]: Yes, I will correct this. Font and color can be set in IB, but my UILabel letter spacing extension does not work well in IB as well as it works in code. So, probably, that is the reason why I left font and color setter in code.

File: ZeplinTask/EditViewController.swift [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    struct CellModel {
+
+        let name: String
+        let number = "8545"
+    }
+
+    let pictureString = "Image-1"
+
+    let arrayTotal = [
+                        CellModel(name: "Total workouts"),
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: @Cona7 Wrong indentation

File: ZeplinTask/EditViewController.swift [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+class EditViewController: UITableViewController {
+
+    struct CellModel {
+
+        let name: String
+        let number = "8545"
+    }
+
+    let pictureString = "Image-1"
```
Comment (Rep2) [inline ZeplinTask/EditViewController.swift]: Think this is unused

**Review by Rep2 — CHANGES_REQUESTED** [review]

File: ZeplinTask/EditViewController.swift (hunk 11 above)
Comment (Cona7) [reply ZeplinTask/EditViewController.swift]: Here's your example you mentioned above:

```
[
    CellModel(name: "Total workouts"),
    CellModel(name: "Total reps")
]
```
I missed one space, but i still don't understand,
do I need to achieve it like this:
```
let arrayMuscles =
    [
        CellModel(name: "Triceps"),
        CellModel(name: "Biceps"),
        CellModel(name: "Bench")
    ]
```

Comment (Rep2) [reply ZeplinTask/EditViewController.swift]: Like this:

```Swift
    let arrayMuscles = [
        CellModel(name: "Triceps"),
        CellModel(name: "Biceps"),
        CellModel(name: "Bench")
    ]
```

Your indentation was like this

```Swift
    let arrayMuscles = [
                          CellModel(name: "Triceps"),
                          CellModel(name: "Biceps"),
                          CellModel(name: "Bench")
                       ]
```

**Review by Rep2 — APPROVED** [review]

**Pull request merged**